src/hotspot/share/runtime/javaCalls.hpp

```diff
@@ -103,23 +103,6 @@ class JavaCallArguments : public StackObj {
     _start_at_zero = false;
   }
 
-  // Helper for push_oop and the like.  The value argument is a
-  // "handle" that refers to an oop.  We record the address of the
-  // handle rather than the designated oop.  The handle is later
-  // resolved to the oop by parameters().  This delays the exposure of
-  // naked oops until it is GC-safe.
```

Comment:
I thought this was the reason we had to have this ugliness.  Once you push an oop to the argument stack, you could have a GC and this was a naked oop.  By having the Handle pointer or jobject pointer, you'd get a pointer to the oop for GC to process.  Maybe this change does the same thing though, only a lot nicer. Yes, it looks like it does.  Now to have Kim point out why it's wrong :(  I hope not because I like this change.

Reply:
The patch does the same thing as previously, but it skips casting oop*/jobject to oop.

Reply:
Ok!